src/wazuh_mcp_server/api/wazuh_client.py

```
@@ -1,568 +1,191 @@
+        except httpx.HTTPStatusError as e:
+            if e.response.status_code == 401:
+                # Token might be expired, try to re-authenticate
+                self.token = None
+                await self._authenticate()
+                # Retry the request once
+                headers = {"Authorization": f"Bearer {self.token}"}
+                response = await self.client.request(method, url, headers=headers, **kwargs)
+                response.raise_for_status()
+                return response.json()
+            else:
+                raise ValueError(f"Wazuh API request failed: {e.response.status_code} - {e.response.text}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)